docs/CHANGELOG/CHANGELOG-1.12.md

```diff
@@ -33,6 +33,196 @@
+
+In the daily operation and maintenance of the Karmada system, many Karmadactl commands are frequently used. 
+
+With this release, Karmadactl now supports autocompletion, which reduces typing significantly and enhances both efficiency and user experience in operating and maintaining Karmadactl. 
+
+Currently, Karmadactl supports command completion for both Bash and Zsh. More information about `karmadactl completion` can be obtained using `karmadactl completion --help`.
+
+(Feature contributors: @zhzhuang-zju)
+
+## Other Notable Changes
+### API Changes
+- Introduced `SecretRef` to `Karmada` API as part of the configuration for connecting to an external etcd cluster can be used to reference a secret that contains credentials for connecting to an external etcd cluster. ([#5699](https://github.com/karmada-io/karmada/pull/5699), @jabellard)
+- Introduced `extraVolumes` and `extraVolumemounts` to the `Karmada` API to optionally specify extra volumes and volume mounts for the Karmada API server component. ([#5509](https://github.com/karmada-io/karmada/pull/5509), @jabellard)
+- Introduced `ApiServerService` field to `Karmada` API as part of the Karmada instance status can be used to reference the API Server service for that instance. This is useful for scenarios where higher level operators need to discover the API Server service of a Karmada instance  for tasks like setting up ingress traffic. ([#5775](https://github.com/karmada-io/karmada/pull/5775), @jabellard)
+- Introduced `CustomCertificate.ApiServerCACert` field to `Karmada` API as part of the `Karmada` spec to specify the reference to a secret that contains a custom CA certificate for the Karmada API Server. ([#5842](https://github.com/karmada-io/karmada/pull/5842), @jabellard)
+- API change: The ServiceType of Karmada API server now has been restrict to `ClusterIP`, `NodePort` and `LoadBalancer`. ([#5769](https://github.com/karmada-io/karmada/pull/5581), @RainbowMango)
+- Introduced a new condition `CompleteAPIEnablements` to represent api collection status of clusters. ([#5400](https://github.com/karmada-io/karmada/pull/5400), @whitewindmills)
+- Introduced `PreserveResourcesOnDeletion` field to both PropagationPolicy and ClusterPropagationPolicy API, which provides the ability to roll back migration safely. ([#5575](https://github.com/karmada-io/karmada/pull/5575), @RainbowMango)
+- API Change: Introduced `FieldOverrider` to both OverridePolicy and ClusterOverridePolicy, which provides the ability to override structured data nested in manifest like ConfigMap or Secret. ([#5581](https://github.com/karmada-io/karmada/pull/5581), @RainbowMango)
+- Introduced `PurgeMode` to GracefulEvictionTask in ResourceBinding. ([#5816](https://github.com/karmada-io/karmada/pull/5816), @mszacillo)
+- Introduced `StatePreservation` to `PropagationPolicy`, which will be used to preserve status in case of application failover. ([#5885](https://github.com/karmada-io/karmada/pull/5885), @RainbowMango)
```

Comment:
> `PropagationPolicy`

Does the CRD resource name in other places need to be unified if quotation marks are not added?

Reply:
Thanks for pointing this out, I think quotation marks could be added uniformly